reference/orders.v2.oas2.yml

```diff
@@ -3763,9 +3763,61 @@ components:
         - product_id
         - product_options
         - quantity
-    orderShipment_Post:
+    orderShipment_Post_Request:
       type: object
-      title: orderShipment_Post
+      title: orderShipment_Post_Request
+      properties:
+        order_address_id:
+          description: ID of the desired `shipping_address` associated with the shipment.
+          example: 20
+          type: integer
+        tracking_number:
+          description: Tracking number of the shipment.
+          example: w4se4b6ASFEW4T
+          type: string
+        shipping_method:
+          description: |
+            Additional information to describe the method of shipment (ex. Standard, Ship by Weight, Custom Shipment). Can be used for live quotes from certain shipping providers.
+            If different from `shipping_provider`, `shipping_method` should correspond to `tracking_carrier`.
+          example: Ship by Weight
+          type: string
+        shipping_provider:
+          type: string
+          description: Enum of the BigCommerce shipping-carrier integration/module.
+          enum:
+            - auspost
+            - canadapost
+            - endicia
+            - usps
+            - fedex
+            - ups
+            - upsready
+            - upsonline
+            - shipperhq
+        tracking_carrier:
+          type: string
+          title: Tracking Carrier
+          description: |-
+            Tracking carrier for the shipment.
+            Acceptable values include an empty string (`""`) or one of the valid tracking-carrier values viewable [here](https://docs.google.com/spreadsheets/d/1w9c_aECSCGyf-oOrvGeUniDl-ARGKemfZl0qSsav8D4/pubhtml?gid=0&single=true) and downloadable as a .CSV file [here](https://docs.google.com/spreadsheets/d/1mTueEynfcEmwsU2y2Jd2MX-8GKwNZrmlRMBcIElg9aY/pub?gid=0&single=true&output=csv).
+        comments:
+          description: Comments the shipper wishes to add.
+          type: string
+        items:
+          description: 'The items in the shipment. This object has the following members, all integer: order_product_id (required), quantity (required), product_id (read-only). A sample items value might be: [ {"order_product_id":16,"product_id": 0,"quantity":2} ]'
+          type: array
+          items:
+            type: object
+            properties:
+              order_product_id:
+                type: integer
+                example: 5
+              quantity:
+                type: number
+                example: 2
+    orderShipment_Post_Response:
```

Comment:
Does it need to be called orderShipment_Put_Request instead?

Reply:
I don't think so, because this component relates to the response body that is referenced on line 679:

`$ref: '#/components/schemas/orderShipment_Post_Response'`.

Currently we only have one component **orderShipment_Post_Response**, so I split it into two components to address the **product_id** field not being required in the request body but should continue to be returned in the response body.

Reply:
Isn't line 679 the PUT request body? 🤔 
<img width="596" alt="Screen Shot 2021-03-11 at 3 47 42 pm" src="https://user-images.githubusercontent.com/12002605/110737310-40885280-8281-11eb-9554-fcfb319b927d.png">

Reply:
You're right, good pick up I didn't see that, thank you!

Changed the approach after also exploring the PUT. Noticed that the expected request body for POST and PUT are different so updated the **orderShipment_Post** component by removing the **product_id** field.

And introduced an **orderShipment_Put** component that does not contain the **items** object.

Reply:
Awesome. Thanks Roger 👍